app/models/manageiq/providers/vmware/infra_manager/inventory/collector/property_collector.rb

```diff
@@ -233,75 +44,35 @@ def prop_set
     end
   end
 
-  def folder_to_child_entity
-    RbVmomi::VIM.TraversalSpec(
-      :name => 'tsFolder', :type => 'Folder', :path => 'childEntity',
-      :selectSet => [
-        RbVmomi::VIM.SelectionSpec(:name => 'tsFolder'),
-        RbVmomi::VIM.SelectionSpec(:name => 'tsDcToDsFolder'),
-        RbVmomi::VIM.SelectionSpec(:name => 'tsDcToHostFolder'),
-        RbVmomi::VIM.SelectionSpec(:name => 'tsDcToNetworkFolder'),
-        RbVmomi::VIM.SelectionSpec(:name => 'tsDcToVmFolder'),
-        RbVmomi::VIM.SelectionSpec(:name => 'tsCrToHost'),
-        RbVmomi::VIM.SelectionSpec(:name => 'tsCrToRp'),
-        RbVmomi::VIM.SelectionSpec(:name => 'tsRpToRp'),
-        RbVmomi::VIM.SelectionSpec(:name => 'tsRpToVm')
-      ]
-    )
-  end
-
-  def datacenter_to_datastore_folder
-    RbVmomi::VIM.TraversalSpec(
-      :name => 'tsDcToDsFolder', :type => 'Datacenter', :path => 'datastoreFolder',
-      :selectSet => [RbVmomi::VIM.SelectionSpec(:name => 'tsFolder')]
-    )
-  end
-
-  def datacenter_to_host_folder
-    RbVmomi::VIM.TraversalSpec(
-      :name => 'tsDcToHostFolder', :type => 'Datacenter', :path => 'hostFolder',
-      :selectSet => [RbVmomi::VIM.SelectionSpec(:name => 'tsFolder')]
-    )
-  end
-
-  def datacenter_to_network_folder
-    RbVmomi::VIM.TraversalSpec(
-      :name => 'tsDcToNetworkFolder', :type => 'Datacenter', :path => 'networkFolder',
-      :selectSet => [RbVmomi::VIM.SelectionSpec(:name => 'tsFolder')]
-    )
-  end
-
-  def datacenter_to_vm_folder
-    RbVmomi::VIM.TraversalSpec(
-      :name => 'tsDcToVmFolder', :type => 'Datacenter', :path => 'vmFolder',
-      :selectSet => [RbVmomi::VIM.SelectionSpec(:name => 'tsFolder')]
-    )
+  def root_folder_traversal_spec(root)
+    RbVmomi::VIM.ObjectSpec(:obj => root, :selectSet => root_folder_select_set)
   end
 
-  def compute_resource_to_host
-    RbVmomi::VIM.TraversalSpec(
-      :name => 'tsCrToHost', :type => 'ComputeResource', :path => 'host',
-    )
-  end
-
-  def compute_resource_to_resource_pool
-    RbVmomi::VIM.TraversalSpec(
-      :name => 'tsCrToRp', :type => 'ComputeResource', :path => 'resourcePool',
-      :selectSet => [RbVmomi::VIM.SelectionSpec(:name => 'tsRpToRp')]
+  def root_folder_select_set
+    traversal_spec_from_hash(
```

Comment:
Perhaps create a `traversal_spec_from_file` method, and have it take "root_folder" as a parameter?  Then the file manipulation stuff doesn't need duplication either

Reply:
:+1: good idea, done